Bump CMake Minimum to 3.5

# Problem
CMake 4.0 is out, and now the minimum required compatibility is 3.5. This should be no problem if y'alls packages are built with the latest Ubuntu with github workflows

Whenever configuring `mcpelauncher-manifest` with cmake `4.0.0`, I get the following error:
```
CMake Error at CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.5 has been removed from CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.

  Or, add -DCMAKE_POLICY_VERSION_MINIMUM=3.5 to try configuring anyway.
```

After setting `-DCMAKE_POLICY_VERSION_MINIMUM=3.5`, I get this new error
```
CMake Error at mcpelauncher-core/CMakeLists.txt:32 (target_link_libraries):
  The keyword signature for target_link_libraries has already been used with
  the target "mcpelauncher-core".  All uses of target_link_libraries with a
  target must be either all-keyword or all-plain.

  The uses of the keyword signature are here:

   * mcpelauncher-core/CMakeLists.txt:11 (target_link_libraries)
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.0)
+cmake_minimum_required(VERSION 3.5)
 
 enable_language(C CXX ASM)
 

